packages/material-ui-unstyled/src/SwitchUnstyled/switchUnstyledClasses.ts

```diff
@@ -0,0 +1,40 @@
+import generateUtilityClass from '../generateUtilityClass';
+import generateUtilityClasses from '../generateUtilityClasses';
+
+export interface SwitchUnstyledClasses {
+  /** Class applied to the root element. */
+  root: string;
+  /** Class applied to the internal input element */
+  input: string;
+  /** Class applied to the thumb element */
+  thumb: string;
+  /** Class applied to the root element if the switch is checked */
```

Comment:
Because:

<img width="482" alt="Capture d’écran 2021-06-29 à 12 44 51" src="https://user-images.githubusercontent.com/3165635/123784525-d222db00-d8d7-11eb-995b-35658b993b4e.png">

Reply:
Sorry but I don't get it - it's still an ordinary CSS class, isn't it? A term "pseudo-class" has a specific meaning in CSS and if we call normal classes this way, I can't see anything but confusion.

Reply:
The class, e.g; `.Mui-checked` can't be styled directly with a CSS selector. If a developer does it, he would override all the others components. In the docs, we have used the term [pseudo-class](https://material-ui.com/customization/components/#pseudo-classes) to signal the difference in behavior. My reasoning was that it behaves differently, therefore it needs to have something different in the description. Happy with an alternative terminology description.

Reply:
Yet again naming things proves to be the [hardest part of programming](https://twitter.com/secretGeek/status/7269997868) :)

You're absolutely right that we should differentiate these classes from the ones that always apply. My concern is with the term pseudo-class and it's overloaded meaning. If a new developer (or even me for that matter) comes in and sees it, they might associate it with CSS pseudo-classes prefixed by a colon and start (unsuccessfully) authoring styles using the same pattern. Since these classes are applied _conditionally_, perhaps "conditional classes" would be a better term?

Reply:
First, to be clear, this thread is off-topic to the effort of the PR. I have started it as the context seems relevant.

The BEM convention has something close: the modifier, but we probably can't use it because readOnly is also a modifier but doesn't get an extra layer of CSS specificity. If we go back to why this exists, it's about allowing developers to customize the baseline style, without messing with the common state styles. "state classes" could be another option but maybe too generic. I like "conditional classes", it's unique, and quick close to the underlying problem. @mnajdova Should we open an RFC/PR to do the change in the codebase?

Reply:
RFC created: https://github.com/mui-org/material-ui/issues/27128